DuplicateFinderEngine/DatabaseHelper.cs

```diff
@@ -24,7 +25,8 @@ static class DatabaseHelper {
 
 			var st = Stopwatch.StartNew();
 			using (var file = new FileStream(path.FullName, FileMode.Open)) {
-				videoFiles = Serializer.Deserialize<List<VideoFileEntry>>(file);
+				videoFiles = Serializer.Deserialize<List<VideoFileEntry>>(file)
```

Comment:
Deserializing a list and casting it to dictionary immediately isn't good. Like the rest there is too much casting where arrays get recreated. Perhaps we might to change the database to dictionary once at all and provide a one time converter for older database ?

Reply:
It's not a cast, it's just a LINQ transform. not sure why you see a problem there. Serializing as a dictionary would just waste space, since the path is in the object anyway.

Reply:
I was talking about the ToDictionary and multiple ToList in your changes. ToList and ToDictionary creates a new array. For small databases it's okay but we don't know how large it will be thus it can either have a small performance / memory usage impact or a heavy.

Reply:
Well the contents are VideoFileEntry so they are a objects on the heap. Those are not moved or recreated when building the dictonary. A List only stores pointers to them, on a 64bit system that is 8byte per item. So a database with a million entries, weighs in at 8 megabytes for the list itself. I'm sure it would be slower to to deserialize with the strings duplicated (once as key and once inside the object). A modern processor moves data on the RAM with several 10GB per second. For comparison everytime a new ffmpeg/ffprobe process is started, atleast 70MB have to be copied in RAM just to run the exe+dlls.